src/history/recorder.test.js

```diff
@@ -58,66 +56,157 @@ describe('Recorder', () => {
+      expect(fs.readFileSync(EXPECTED_FILE_PATH, { encoding: 'utf8' })).to.equal(FILE_CONTENT);
+    });
+
+    it('returns the file path', () => {
+      expect(path).to.equal(EXPECTED_FILE_PATH);
+    });
+
+    it('returns the commit id', () => {
+      expect(commit.hash).to.include(id);
     });
 
-    it('commits the file for the given service', () => {
-      expect(id).to.exist;
-      expect(id).to.be.a('string');
+    it('saves the changelog in the commit message', () => {
+      expect(commit.message).to.equal(changelog);
     });
+  });
+
+  describe('#getPathFor', () => {
+    it('returns the file path for the given service provider’s document type', async () => {
+      expect(subject.getPathFor(SERVICE_PROVIDER_ID, TYPE)).to.equal(EXPECTED_FILE_PATH);
+    });
+  });
 
-    context('when this is the first record', () => {
-      it('returns a boolean to specify this is the first one', () => {
-        expect(isFirstRecord).to.equal(true);
+  describe('#isTracked', () => {
+    after(resetGitRepository);
+
+    context('when the file does not exists', () => {
+      it('returns false', async () => {
+        expect(await subject.isTracked(SERVICE_PROVIDER_ID, TYPE)).to.be.false;
       });
     });
 
-    context('when this is not the first record', () => {
-      it('returns a boolean to specify this is not the first one', async () => {
-        const recordResult = await subject.record({
+    context('when the file already exists', () => {
+      before(async () => {
+        await subject.record({
           serviceId: SERVICE_PROVIDER_ID,
           documentType: TYPE,
-          content: PERSIST_FILE_CONTENT,
-          isFiltered: false
+          content: FILE_CONTENT,
+          changelog: 'Start tracking document'
         });
-        expect(recordResult.isFirstRecord).to.equal(false);
       });
+
+      it('returns true', async () => {
+        expect(await subject.isTracked(SERVICE_PROVIDER_ID, TYPE)).to.be.true;
+      });
+    });
+  });
+
+  describe('#_getCommits', () => {
+    let firstRecordId;
+    let secondRecordId;
+    before(async () => {
+      let record = await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: FILE_CONTENT,
+      });
+      firstRecordId = record.id;
+      record = await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: `${FILE_CONTENT} (with additional content to trigger a record)`,
+      });
+      secondRecordId = record.id;
+    });
+
+    after(resetGitRepository);
+
+    it('returns all commits related to the given file path', async () => {
+      const filePath = subject.getPathFor(SERVICE_PROVIDER_ID, TYPE);
+      const commits = await subject._getCommits(filePath);
+      expect(commits).to.have.length(2);
+      expect(commits[0].hash).to.include(secondRecordId);
+      expect(commits[1].hash).to.include(firstRecordId);
+    });
+  });
```

Comment:
We should record at least another entry for another document, otherwise `_getCommits` might just as well be returning the entire history and we wouldn't know 😅